engine/src/main/java/org/pentaho/di/trans/steps/script/ScriptAddedFunctions.java

```diff
@@ -0,0 +1,2532 @@
+
+          if ( strType.equals( "y" ) ) {
+            return new Double( endDate.get( Calendar.YEAR ) - startDate.get( Calendar.YEAR ) );
+          } else if ( strType.equals( "m" ) ) {
+            int iMonthsToAdd = ( endDate.get( Calendar.YEAR ) - startDate.get( Calendar.YEAR ) ) * 12;
+            return new Double( ( endDate.get( Calendar.MONTH ) - startDate.get( Calendar.MONTH ) ) + iMonthsToAdd );
+          } else if ( strType.equals( "d" ) ) {
+            return new Double( ( ( endL - startL ) / 86400000 ) );
+          } else if ( strType.equals( "wd" ) ) {
+            int iOffset = -1;
+            if ( endDate.before( startDate ) ) {
+              iOffset = 1;
+            }
+            while ( ( iOffset == 1 && endL < startL ) || ( iOffset == -1 && endL > startL ) ) {
+              int day = endDate.get( Calendar.DAY_OF_WEEK );
+              if ( ( day != Calendar.SATURDAY ) && ( day != Calendar.SUNDAY ) ) {
+                iRC++;
+              }
+              endDate.add( Calendar.DATE, iOffset );
+              endL = endDate.getTimeInMillis() + endDate.getTimeZone().getOffset( endDate.getTimeInMillis() );
+            }
+            return new Double( iRC );
+          } else if ( strType.equals( "w" ) ) {
+            int iDays = (int) ( ( endL - startL ) / 86400000 );
+            return new Double( iDays / 7 );
+          } else if ( strType.equals( "ss" ) ) {
+            return new Double( ( ( endL - startL ) / 1000 ) );
+          } else if ( strType.equals( "mi" ) ) {
+            return new Double( ( ( endL - startL ) / 60000 ) );
+          } else if ( strType.equals( "hh" ) ) {
+            return new Double( ( ( endL - startL ) / 3600000 ) );
+          } else {
+            return new Double( ( ( endL - startL ) / 86400000 ) );
+          }
+          /*
+           * End Bugfix
+           */
+        }
+      } catch ( Exception e ) {
+        throw new RuntimeException( e.toString() );
+      }
+    } else {
+      throw new RuntimeException( "The function call dateDiff requires 3 arguments." );
+    }
+  }
+
+  public static Object getNextWorkingDay( ScriptEngine actualContext, Bindings actualObject, Object[] ArgList,
+    Object FunctionContext ) {
+    // (Date dIn){
+    if ( ArgList.length == 1 ) {
+      try {
+        if ( isNull( ArgList[0] ) ) {
+          return null;
+        } else if ( isUndefined( ArgList[0] ) ) {
+          return undefinedValue;
+        }
+        java.util.Date dIn = (java.util.Date) ArgList[0];
+        Calendar startDate = Calendar.getInstance();
+        startDate.setTime( dIn );
+        startDate.add( Calendar.DATE, 1 );
+        while ( startDate.get( Calendar.DAY_OF_WEEK ) == Calendar.SATURDAY
+          || startDate.get( Calendar.DAY_OF_WEEK ) == Calendar.SUNDAY ) {
+          startDate.add( Calendar.DATE, 1 );
+        }
+        return startDate.getTime();
+      } catch ( Exception e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)